cpp/src/io/orc/reader_impl.cu

```diff
@@ -781,6 +781,12 @@ table_with_metadata reader::impl::read(int skip_rows, int num_rows, int stripe,
                                              out_buffers[i], stream, _mr));
       }
     }
+  } else {
```

Comment:
The corresponding `if` is ~150 lines away, which made this a little tough to read

Reply:
I don't disagree, but I did this to avoid inverting the logic to a possibly-confusing `if (not <condition>) { /*make empty cols*/ }` block at the top. I'll defer to @OlivierNV's judgement here.

Reply:
What about...
```
auto needs_to_be_striped = num_rows > 0 && selected_stripes.size() != 0

if (not needs_to_be_striped) {
    ....
} else {
    ....
}
```

Reply:
ok, I moved the check to the top